libs/core-ui/src/lib/util/calculateBoxData.ts

```diff
@@ -1,44 +1,59 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+import { ErrorCohort } from "../Cohort/ErrorCohort";
 import { IHighchartBoxData } from "../Interfaces/IHighchartBoxData";
 
-export function calculateBoxData(data: number[]): IHighchartBoxData {
-  const q1 = getPercentile(data, 25);
-  const median = getPercentile(data, 50);
-  const q3 = getPercentile(data, 75);
-  const iqr = q3 - q1;
-  const lowerFence = q1 - iqr * 1.5;
-  const upperFence = q3 + iqr * 1.5;
-  const outliers = [];
+export function calculateBoxPlotDataFromErrorCohort(
+  errorCohort: ErrorCohort,
+  index: number,
+  key: string
+) {
+  // key is the identifier for the column (e.g., probability)
+  return calculateBoxPlotData(
+    errorCohort.cohort.filteredData.map((dict) => dict[key]),
+    index
+  );
+}
 
-  for (const datum of data) {
-    if (datum < lowerFence || datum > upperFence) {
-      outliers.push(datum);
-    }
-  }
+export function calculateBoxPlotData(
+  data: number[],
+  index?: number
+): IHighchartBoxData {
+  data.sort((number1: number, number2: number) => {
+    return number1 - number2;
+  });
+  const firstQuartile = getPercentile(data, 25);
+  const median = getPercentile(data, 50);
+  const thirdQuartile = getPercentile(data, 75);
+  const interquartileRange = thirdQuartile - firstQuartile;
+  // calculate fences as min and max allowed values to be inside the box
+  const lowerFence = firstQuartile - interquartileRange * 1.5;
+  const upperFence = thirdQuartile + interquartileRange * 1.5;
+  const nonOutliers = data.filter(
+    (element) => element >= lowerFence && element <= upperFence
+  );
+  const outliers = data.filter(
+    (element) => element < lowerFence || element > upperFence
+  );
   return {
-    lowerPercentile: q1,
-    max: upperFence,
-    mean: mean(data),
+    high: nonOutliers[nonOutliers.length - 1],
+    low: nonOutliers[0],
     median,
-    min: lowerFence,
     outliers,
-    upperPercentile: q3
+    q1: firstQuartile,
+    q3: thirdQuartile,
+    x: index
   };
 }
 
-function getPercentile(data: number[], percentile: number): number {
-  data.sort((a, b) => a - b);
-  const index = (percentile / 100) * data.length;
+function getPercentile(sortedData: number[], percentile: number) {
+  const index = (percentile / 100) * sortedData.length;
+  let result;
   if (Math.floor(index) === index) {
-    return (data[index - 1] + data[index]) / 2;
+    result = (sortedData[index - 1] + sortedData[index]) / 2;
```

Comment:
if the chart only contained a single point, might this break?  wondering if we need to make sure sortedData length > 1 in this case?  Not really sure, maybe it's not relevant considering the code outside this method.
```
sortedData[index - 1] + sortedData[index]
```

Reply:
I wonder if there are cases where someone might create a cohort with a single point and this might break

Reply:
I'm writing unit tests for this now :-)

Reply:
FWIW if there's only 1 then you go into the other case (`else`).